app/src/main/java/org/mozilla/focus/broadcastreceiver/DownloadBroadcastReceiver.java

```diff
@@ -0,0 +1,105 @@
+    public void onReceive(Context context, Intent intent) {
+        final long downloadReference = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, -1);
+        displaySnackbar(context, downloadReference, downloadManager);
+    }
+
+    private void displaySnackbar(final Context context, long completedDownloadReference, DownloadManager downloadManager) {
+        if (!isFocusDownload(completedDownloadReference)) {
+            return;
+        }
+
+        final DownloadManager.Query query = new DownloadManager.Query();
+        query.setFilterById(completedDownloadReference);
+        try (Cursor cursor = downloadManager.query(query)) {
+            if (cursor.moveToFirst()) {
+                int statusColumnIndex = cursor.getColumnIndex(DownloadManager.COLUMN_STATUS);
+                if (DownloadManager.STATUS_SUCCESSFUL == cursor.getInt(statusColumnIndex)) {
+                    String uriString = cursor.getString(cursor.getColumnIndex(DownloadManager.COLUMN_LOCAL_URI));
+
+                    final String localUri = uriString.startsWith(FILE_SCHEME) ? uriString.substring(FILE_SCHEME.length()) : uriString;
+                    final String fileExtension = MimeTypeMap.getFileExtensionFromUrl(localUri);
+                    final String mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(fileExtension);
+                    final String fileName = URLUtil.guessFileName(localUri, null, mimeType);
+
+                    final File file = new File(localUri);
+                    final Uri uriForFile = FileProvider.getUriForFile(context, BuildConfig.APPLICATION_ID + FILE_PROVIDER_EXTENSION, file);
+
+                    final Intent openFileIntent = IntentUtils.createOpenFileIntent(uriForFile, mimeType);
+                    showSnackbarForFilename(openFileIntent, context, fileName);
+                }
+            }
+        }
+        removeFromHashSet(completedDownloadReference);
+    }
+
+    private void showSnackbarForFilename(final Intent openFileIntent, final Context context, String fileName) {
+        final Snackbar snackbar = Snackbar
+                .make(browserContainer, String.format(context.getString(R.string.download_snackbar_finished), fileName), Snackbar.LENGTH_LONG);
+        if (IntentUtils.activitiesFoundForIntent(context, openFileIntent)) {
+            snackbar.setAction(context.getString(R.string.download_snackbar_open), new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    context.startActivity(openFileIntent);
+                }
+            });
+            snackbar.setActionTextColor(ContextCompat.getColor(context, R.color.download_dialog_text));
+        }
+        snackbar.show();
+    }
+
+    private boolean isFocusDownload(long completedDownloadReference) {
+        return (queuedDownloadReferences.contains(completedDownloadReference));
+    }
+
+    private void removeFromHashSet(long completedDownloadReference) {
```

Comment:
nit: I don't think this warrants a separate helper method. `hashSet.remove(thingie)` is already pretty readable and to which hash set does "removeFromHashSet" refer to?